File: src/Dashboard/Data/Logs/IndexerLogEntry.cs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System;

namespace Dashboard.Data.Logs
{
    public class IndexerLogEntry 
    {
        public string Id { get; set; }

        public DateTime Date { get; set; }

        public string Title { get; set; }

        public string ExceptionDetails { get; set; }
    }
}

File: src/Dashboard/app/models/IndexerLogEntry.js
angular.module('dashboard').factory('IndexerLogEntry', function (stringUtils) {
    function IndexerLogEntry() {

    }

    IndexerLogEntry.fromJson = function (item) {
        var model = new IndexerLogEntry();

        model.id = item.id;
        model.date = item.date;
        model.title = item.title;
        model.exceptionDetails = item.exceptionDetails;

        return model;
    };

    return IndexerLogEntry;
});
